specs/language/resources.tex

```diff
@@ -0,0 +1,755 @@
+\Ch{Resource Types}{Resources}
+
+Resources are built-in intangible types representing memory with an external interface.
+
+These take the form of Typed Buffers, Raw Buffers, Textures, and Samplers.
+Buffer and Texture types can be read-only or writable.
+
+\Sec{Typed Buffers}{Resources.tybufs}
+
+The typed buffer class template represents a one-dimensional resource containing an array of a single given type.
+Its contents are indexed by typed access to each element
+Types may have their formats converted upon load.
+
+Template types can be any type that totals or can be padded to 16 bytes.
+
+All typed buffers can be read through subscript operators or Load methods.
+Writable typed buffers can be written through subscript operators.
+
+Where \texttt{T} can be any arithmetic type \ref{Basic.types.arithmetic}
+or a vector with a maximum of four elements containing such an arithmetic type.
+The total size of a single contained element must be less than 16 bytes.
+
+Typed buffers perform format conversions on load such that the underlying data
+gets converted to the destination type.
+
+\begin{HLSL}
+template <typename T = float4>
+ class Buffer {
+ public:
+   Buffer();
+   Buffer(const Buffer &buf);
+   Buffer operator=(Buffer buf);
+
+   void GetDimensions(out uint width) const;
+
+   // Element access.
+   T operator[](int index) const;
+   T Load(in int index) const;
+   T Load(in int index, out uint status) const;
+};
+
+
+template <typename T>
+ class RWBuffer : public Buffer {
+ public:
+   RWBuffer();
+   RWBuffer(RWBuffer buf);
+   RWBuffer operator=(RWBuffer buf);
+
+   // element access
+   T &operator[](int n);
```

Comment:
Should this be `T &operator[](size_t n);`?

Along the same lines, should the type of `index` argument of `Buffer::Load(...)` methods above be `size_t` instead of `int`?

Reply:
> Should this be `T &operator[](size_t n);`?
> 
> Along the same lines, should the type of `index` argument of `Buffer::Load(...)` methods above be `size_t` instead of `int`?

Never mind. It would contradict the [public API](https://learn.microsoft.com/en-us/windows/win32/direct3dhlsl/sm5-object-buffer-load)